### PR TITLE
prettyprint

### PR DESCRIPTION
**Pull request by hyjorc1**

### CODE REVIEW

**Comment by psybers** [comment]
would it be possible to add test cases for whatever it is you are fixing?

I added a new class, `boa/test/functions/TestPrettyprint.java` with some samples of the test. You can run these tests with `ant -Dtest.case=boa.test.functions.TestPrettyprint test-specific` (they should all pass right now).  Then you can add a test case for each broken thing you are fixing here.  This will a) help people like me review the change (I'm not sure exactly why you are making these changes) and b) ensure future behavior doesnt regress and break your fixes.

**Comment by hyjorc1** [comment]
Sorry, I will add test cases in this and my future work. I am using prettyprint recently in my research. I found some errors in those domain-specific functions because of incorrect attribute invocations. The prettyprint of the lambda expression is also added for further research.

File: src/test/boa/test/functions/TestPrettyprint.java [hunk 1]
```diff
@@ -35,34 +35,82 @@
...
-    }
-
-    @Test()
-    public void testPrettyprint() throws Exception {
-        assertEquals(code, prettyprint(parse(code)));
-    }
+	@Test()
+	public void testPrettyprint() throws Exception {
+		String expected = prettyprint(parse(code));
+//		System.out.println(expected);
```
Comment (psybers) [inline src/test/boa/test/functions/TestPrettyprint.java]: can you remove the commented lines?

File: src/test/boa/test/functions/TestPrettyprint.java [hunk 2]
```diff
@@ -35,34 +35,82 @@
  */
 @RunWith(Parameterized.class)
 public class TestPrettyprint {
-    private final static String CLASS_START = "class c {\n";
-    private final static String CLASS_END = "}\n";
+	private final static String CLASS_START = "class c {\n";
+	private final static String CLASS_END = "}\n";
+	private final static String STATEMENT_START = CLASS_START + "\tvoid m()\n\t{\n\t\t";
```
Comment (psybers) [inline src/test/boa/test/functions/TestPrettyprint.java]: why not use your indent() function here too?

**Pull request merged**